astropy/constants/constant.py

```diff
@@ -159,10 +159,10 @@ def name(self):
         return self._name
 
     @lazyproperty
-    def unit(self):
+    def _unit(self):
```

Comment:
No, not quite so bad: since `Constants` subclasses from `Quantity`, it still has the `unit` property (which just returns `_unit`, i.e., this thing here). I had to make sure, though, that `_unit` was a `Unit` instance, which is what this `lazyproperty` takes care of.